### PR TITLE
[10.x] Add an encoding check string utility and a validator that uses it

### PR DESCRIPTION
**Pull request by Synchro**
Laravel has an `ascii` validator that checks whether a value (string, or other type converted to string) is valid ASCII. However, there are no validators or support methods to check for any other character set (apart from some specific cases like `alpha_num`).

The mbstring extension provides the [`mb_check_encoding`](https://www.php.net/manual/en/function.mb-check-encoding.php) function that will check whether a string is valid in a given encoding. In the same approach to other methods in the `Str` support class, I have wrapped this function to work the same way as the `isAscii` method.

This method allows you to check whether a string is valid in a given encoding:

    $isValid = Str::isEncoding("\xE9", 'ISO-8859-1'); //é in ISO-8859-1

The selected encoding is checked before use using the [`mb_list_encodings`](https://www.php.net/manual/en/function.mb-list-encodings.php) function; a non-existent encoding will result in a `false` return value. An empty string will always return `true` (same behaviour as the `isAscii` method).

I've added a validator that uses this new method as a validator using the `encoding` name, so you can define a validator rule like:

    'foo' => 'encoding:UTF-8',

I wrote this because I needed to validate UTF-8 input where it was quite likely to receive text incorrectly encoded as ISO-8859-1 or Windows-1252.

A key limitation of this validator (it's an inherent limitation of 8-bit charsets and UTF-8), is that it can't detect when a UTF-8 string has been presented as ISO-8859-\*. This is because all 8-bit values are valid in ISO-8859 charsets, and consequently all valid UTF-8 is also valid ISO-8859-\*, but the reverse is not true. It's possible to partially detect this through heuristics, but that's beyond the scope of this utility.

TL;DR: This PR adds a `Str::isEncoding` method and an `encoding` validator that uses it, along with tests for both uses.

### CODE REVIEW

**Comment by taylorotwell** [comment]
Feel free to write a custom validator for this. 👍
